scripts/mbedtls_framework/config_common.py

```diff
@@ -0,0 +1,426 @@
+
+        If value is None and name is already known, don't change its value.
+        If value is None and name is not known, set its value.
+        """
+        if name in self.settings:
+            setting = self.settings[name]
+            if setting.value != value or not setting.active:
+                setting.configfile.modified = True
+            if value is not None:
+                setting.value = value
+            setting.active = True
+        else:
+            configfile = self._get_configfile(name)
+            self.settings[name] = Setting(configfile, True, name, value=value)
+            configfile.modified = True
+
+    def unset(self, name):
+        """Make name unset (inactive).
+
+        name remains known if it was known before.
+        """
+        if name not in self.settings:
+            return
+
+        setting = self.settings[name]
+        # Check if modifying the config file
+        if setting.active:
+            setting.configfile.modified = True
+
+        setting.active = False
+
+    def adapt(self, adapter):
+        """Run adapter on each known symbol and (de)activate it accordingly.
+
+        `adapter` must be a function that returns a boolean. It is called as
+        `adapter(name, active, section)` for each setting, where `active` is
+        `True` if `name` is set and `False` if `name` is known but unset,
+        and `section` is the name of the section containing `name`. If
+        `adapter` returns `True`, then set `name` (i.e. make it active),
+        otherwise unset `name` (i.e. make it known but inactive).
+        """
+        for setting in self.settings.values():
+            is_active = setting.active
+            setting.active = adapter(setting.name, setting.active,
+                                     setting.section)
+            # Check if modifying the config file
+            if setting.active != is_active:
+                setting.configfile.modified = True
+
+    def change_matching(self, regexs, enable):
+        """Change all symbols matching one of the regexs to the desired state."""
+        if not regexs:
+            return
+        regex = re.compile('|'.join(regexs))
+        for setting in self.settings.values():
+            if regex.search(setting.name):
+                # Check if modifying the config file
+                if setting.active != enable:
+                    setting.configfile.modified = True
+                setting.active = enable
+
+    def _get_configfile(self, name=None):
+        """Find a config for a setting name.
+
+        If more then one configfile is used this function must be overridden.
+        """
+
+        if name and name in self.settings:
+            return self.get(name).configfile
+        return self.configfiles[0]
+
+    def write(self, filename=None):
+        """Write the whole configuration to the file it was read from.
+
+        If filename is specified, write to this file instead.
```

Comment:
I agree with there should be a list. The real question is do we want to differentiate the single and multiple config handler descendants? The latter is using more then one config file and need to specify all of them or none of them to work. Not to mention the order, it can be easily ending up with swapping the content of the files.

IMO the solution would be using str for single configs handlers and list of tupples or a dict for multiple config handlers.

Reply:
Looking at this, it seems to be the case that writing a configuration associated to the crypto and mbedtls configuration file (class CombinedConfig) is actually working fine and that the only "issue" we have currently revealing that things could maybe be better shaped is the "#pylint: disable=arguments-differ" needed for the CombinedConfig when overwriting the write method?
If that's the case and thus the writing of the two configuration files with the CombinedConfig `write` method is working fine I am happy to leave things as they are. I would only ask for a comment in config.py explaining the "#pylint: disable=arguments-differ" mentioned above.